src/core/thread/mle_router.cpp

```diff
@@ -2502,6 +2511,16 @@ void MleRouter::SetLeaderWeight(uint8_t aWeight)
     mLeaderWeight = aWeight;
 }
 
+uint32_t MleRouter::GetLeaderPartitionId(void) const
+{
+	return mLeaderPartitionId;
+}
+
+void MleRouter::SetLeaderPartitionId(uint32_t aPartitionId)
+{
+	mLeaderPartitionId = aPartitionId;
```

Comment:
we might hope it would return error if set operation is happening after the device joined the thread network?